catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/CachePutPlugin.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.cache;
+
+import ddf.catalog.data.Metacard;
+import java.util.Optional;
+
+public interface CachePutPlugin {
```

Comment:
@pklinef and I discussed several variations. We didn't like PreCachePlugin because it would invite multiple methods being added to the interface (ie. `put`, `delete` etc). We wanted to make sure there was only one method in the interface.